hammer/client.go

```diff
@@ -152,12 +156,20 @@ func (w httpLeafWriter) Write(ctx context.Context, newLeaf []byte) (uint64, erro
 	if err != nil {
 		return 0, fmt.Errorf("failed to read body: %v", err)
 	}
-	if resp.StatusCode != http.StatusOK {
+	switch resp.StatusCode {
+	case http.StatusOK:
+		if resp.Request.Method != http.MethodPost {
+			return 0, fmt.Errorf("write leaf was redirected to %s", resp.Request.URL)
+		}
+		// Continue below
+	case http.StatusServiceUnavailable, http.StatusBadGateway, http.StatusGatewayTimeout:
+		// These status codes may indicate a delay before retrying, so handle that here:
+		retryDelay(resp.Header.Get("RetryAfter"), time.Second)
```

Comment:
This just computes a value and then throws it away, right? Where does the retry happen?

Reply:
d'oh, yes - it used to actually delay in the method.